openc3/lib/openc3/io/posix_serial_driver.rb

```diff
@@ -132,9 +137,19 @@ def read
       begin
         data = @handle.read_nonblock(65535)
       rescue Errno::EAGAIN, Errno::EWOULDBLOCK
-        result = IO.fast_select([@handle], nil, nil, @read_timeout)
-        if result
-          retry
+        begin
+          read_ready, _ = IO.fast_select(@readers, nil, nil, @read_timeout)
+        rescue IOError
+          @pipe_reader.close unless @pipe_reader.closed?
+          return ""
+        end
+        if read_ready
+          if read_ready.include?(@pipe_reader)
+            @pipe_reader.close unless @pipe_reader.closed?
+            return ""
```

Comment:
Why would you close the pipe_reader if the read_ready says it's ready?

Reply:
pipe_reader is only ever ready, if pipe_writer was written.  That means we have been closed.